core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/MarecoAllowance.java

```
@@ -498,19 +502,17 @@ private Envelope binarySearchLoop(DoubleBinarySearch search,
             search.feedback(output);
         }
         if (!search.complete())
-            throw makeMarecoError(search);
+            throwMarecoError(search);
```

you could return the exception as was done previously